### PR TITLE
MAINT: Removed some warnings in tests

### PR DESCRIPTION
**Pull request by gfyoung**
Per discussion with @jreback <a href="https://github.com/pydata/pandas/pull/13671#issuecomment-233508468">here</a>.

### CODE REVIEW

File: pandas/tests/test_categorical.py [hunk 1]
```diff
@@ -2934,26 +2933,28 @@ def test_value_counts_with_nan(self):
             s = pd.Series(pd.Categorical(["a", "b", "a"],
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: this is an incorrect construction, categories should not be ``nan`` by definition

File: pandas/tests/test_categorical.py [hunk 2]
```diff
@@ -4122,8 +4123,9 @@ def f():
         c[0] = np.nan
         df = pd.DataFrame({"cats": c, "vals": [1, 2, 3]})
 
-        cat_exp = Categorical(["a", "b", "a"], categories=["a", "b", np.nan])
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: same here, by definition the cat should not be constructed like this (instead construct from codes and valid categories)

File: pandas/tests/test_categorical.py (hunk 1 above)
Comment (gfyoung) [inline pandas/tests/test_categorical.py]: The test is expecting a `nan` in the construction of the categories if you look at what it is testing.

File: pandas/tests/test_categorical.py (hunk 2 above)
Comment (gfyoung) [inline pandas/tests/test_categorical.py]: Again, the test is expecting a `Categorical` with `NaN` in the categories AFAICT.

File: pandas/tests/test_categorical.py (hunk 1 above)
Comment (jreback) [inline pandas/tests/test_categorical.py]: and that's wrong

Comment (gfyoung) [inline pandas/tests/test_categorical.py]: Ah, okay.  Fair enough.  Done.

**Comment by gfyoung** [comment]
@jreback : it seems like `test_categorical.py` has A LOT of tests where they expect to use `np.nan` as a category.  Does it make sense to modify / remove them?

**Comment by jreback** [comment]
can't be very many as we have a warning for it
I only see a few

**Comment by gfyoung** [comment]
1) A Ctrl+F returns 30+ results (all of them caught with `tm.assert_produces_warnings(FutureWarning)`
2) So is that a yes or no?

**Comment by jreback** [comment]
there are warnings but should not be from that

pls show an example or 2

**Comment by gfyoung** [comment]
1) Constructor tests <a href="https://github.com/gfyoung/pandas/blob/feecb8b59d9417963d3e2945d9c38894bd3834b8/pandas/tests/test_categorical.py#L155">here</a> (just testing that duplicates get rejected)

2) Describing with `NaN` categories <a href="https://github.com/gfyoung/pandas/blob/feecb8b59d9417963d3e2945d9c38894bd3834b8/pandas/tests/test_categorical.py#L612">here</a>

3) Another constructor test <a href="https://github.com/gfyoung/pandas/blob/feecb8b59d9417963d3e2945d9c38894bd3834b8/pandas/tests/test_categorical.py#L1137">here</a>.

4) This entire test <a href="https://github.com/gfyoung/pandas/blob/feecb8b59d9417963d3e2945d9c38894bd3834b8/pandas/tests/test_categorical.py#L1147">here</a>.

Perhaps I'm conflating unnecessary tests with backwards compat tests since the code still has to work with this deprecated behaviour, not sure...

**Comment by jreback** [comment]
yeah these are all fine
when this behavior is removed in the future the tests can be changed

for now since we have enough catching of the warning
prob just fix the examples that were causing a warning (rather than catch them)

**Comment by gfyoung** [comment]
Okay, sounds good.  I'll fix the other one then.

**Comment by gfyoung** [comment]
@jreback : Removed the previous warnings, and Travis is still passing.  Ready to merge if there are no other concerns.

**Comment by jreback** [comment]
ty!

**Pull request closed without merging**